banzai_floyds/tests/test_fringing.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+from banzai_floyds.fringe import FringeMaker, find_fringe_offset, fit_smooth_fringe_spline
+from banzai_floyds.tests.utils import generate_fake_science_frame
+from banzai_floyds.fringe import FringeCorrector
+from banzai import context
+
+
+def test_find_fringe_offset():
+    # Make fringe data using a sin function and our fake data generator
+    np.random.seed(234142)
+    frame = generate_fake_science_frame(flat_spectrum=False, include_sky=False, background=6000.0,
+                                        fringe=True, fringe_offset=5, include_trace=False)
+    x2d, y2d = np.meshgrid(np.arange(frame.shape[1]), np.arange(frame.shape[0]))
+    red_order = frame.orders.data == 1
+    fringe_data = np.ones_like(frame.data)
+    fringe_data[red_order] = np.sin(frame.fringe_wave_number * frame.wavelengths.data[red_order])
+    fringe_data[red_order] = 1.0 + 0.5 * (x2d[red_order] / np.max(x2d[red_order])) * fringe_data[red_order]
+
+    class FakeFrame(object):
+        pass
+
+    fringe_frame = FakeFrame()
+    fringe_frame.data = fringe_data
+    fringe_frame.orders = frame.orders
+    fringe_frame.error = 0.01 * fringe_data
+    fringe_frame.shape = fringe_data.shape
+    # fit the fringe offset
+
+    fringe_spline = fit_smooth_fringe_spline(fringe_data, frame.orders.data == 1)
+    best_fit_offset = find_fringe_offset(frame, fringe_spline)
+    # assert that the offset is correct
+    np.testing.assert_allclose(best_fit_offset, frame.input_fringe_shift, atol=0.2)
+
+
+def test_create_super_fringe():
+    np.random.seed(28159)
+    # Make a set of fake images all with different offsets
+    frames = []
+    # Set the first one to zero to define the reference position of the set
+    frames.append(generate_fake_science_frame(flat_spectrum=False, include_sky=False, fringe=True,
```

Comment:
shouldn't this just be `frames = [generate_fake_science_frame(...)]` and line 38 removed?

Reply:
I kind of like having them separate just to explicitly emphasize that we are prepopulating a first element with no shift.